File: haohaninfo/交易實務案例/56-1.R
#取得即時報價，詳細在技巧51
source("function.R")

#設定變數初始格式
inDesk<-0
outDesk<-0

while(TRUE){
 #取得即時成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 MatchTime <- Mdata[[1]][1]
 MatchPrice <- as.numeric(Mdata[[1]][2])
 MatchQty <- as.numeric(Mdata[[1]][3])
 
 #取得上下五檔價資訊
 UpDn5data<-GetUpDn5Data(DataPath,Date) 
 Dn1Price<- as.numeric(UpDn5data[[1]][2])
 Up1Price<- as.numeric(UpDn5data[[1]][12])

 #進行內外盤判斷
 if ( MatchPrice >= Up1Price ){
  outDesk <- outDesk+MatchQty
 }else if( MatchPrice <= Dn1Price){
  inDesk <- inDesk+MatchQty
 }

 print(paste(MatchTime,MatchPrice,outDesk,inDesk))

}

File: haohaninfo/即時範例/56-1.py
# -*- coding: UTF-8 -*-

#取得報價資訊，詳情請查看技巧51
execfile('function.py')

#定義指標變數
OutDesk=0
InDesk=0

#取得成交資訊
for i in getMatch():		
  MatchInfo=i.split(',')
  MatchTime=MatchInfo[0]
  MatchPrcie=int(MatchInfo[1])
  MatchQty=int(MatchInfo[2])
  #取得上下五檔價資訊
  UpDn5Info=getLastUpDn5()
  Dn1Price=int(UpDn5Info[1])
  Up1Price=int(UpDn5Info[11])

  #進行內外盤判斷
  if MatchPrcie>=Up1Price:
   OutDesk+=MatchQty
  if MatchPrcie<=Dn1Price:
   InDesk+=MatchQty
   
  print MatchTime,"OutDesk",OutDesk,"InDesk",InDesk
